x-pack/elastic-agent/control.proto

```diff
@@ -0,0 +1,118 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+syntax = "proto3";
+
+package proto;
+
+option cc_enable_arenas = true;
+option go_package = "pkg/agent/control/proto;proto";
+
+// Status codes for the current state.
+enum Status {
+  STARTING = 0;
+  CONFIGURING = 1;
+  HEALTHY = 2;
+  DEGRADED = 3;
+  FAILED = 4;
+  STOPPING = 5;
+  UPGRADING = 6;
+}
+
+// Action status codes for restart and upgrade response.
+enum ActionStatus {
+  // Action was successful.
+  SUCCESS = 0;
+  // Action failed.
+  FAILURE = 1;
+}
+
+// Empty message.
+message Empty {
+}
+
+// Version response message.
+message VersionResponse {
+  // Current running version.
+  string version = 1;
+  // Current running commit.
+  string commit = 2;
+  // Current running build time.
+  string buildTime = 3;
+  // Current running version is a snapshot.
+  bool snapshot = 4;
+}
+
+message RestartResponse {
+  // Response status.
+  ActionStatus status = 1;
+  // Error message when it fails to trigger restart.
+  string error = 2;
+}
+
+// Upgrade request message.
+message UpgradeRequest {
+  // (Optional) Version to upgrade to.
+  //
+  // If not provided Elastic Agent will auto discover the latest version in the same major
```

Comment:
this makes sense but i'm worried a bit about sealed env. while it initially might all work without even using default sourceURI as everything is packed. then on upgrade request without version and sourceURI it will fail. 
i would be up for these to values being required and relaxing later when we have clear use cases. relaxing is not breaking other way around is though

Reply:
Being that CLI call to upgrade over the socket will return an `UpgradeResponse` we can place in the error message why the upgrade cannot be performed. Maybe it is we cannot discover the next version, or maybe we make the version required in some cases like when on the last minor of the current major.